src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -274,12 +274,24 @@ public bool TrySkipToAny(ReadOnlySpan<T> delimiters, bool advancePastDelimiter =
         /// Skip consecutive instances of the given <paramref name="value"/>.
         /// </summary>
         /// <returns>True if any Ts were skipped.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public bool SkipPast(T value)
         {
             int start = Consumed;
-            while (!End && CurrentSpan[CurrentSpanIndex].Equals(value))
+            ReadOnlySpan<T> unread = UnreadSpan;
```

Comment:
@JeremyKuhne, please review the changes made to BufferReader_search (mainly `SkipPast{Any}`).